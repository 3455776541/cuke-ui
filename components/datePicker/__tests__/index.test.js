import React from "react";
import assert from "power-assert";
import { render, shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import DatePicker from "../index";
import Button from "../../button";

describe("<DatePicker/>", () => {
  it("should render a <DatePicker/> components", () => {
    const wrapper = render(<DatePicker />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should render find .cuke-date-picker classnames", () => {
    const wrapper = shallow(<DatePicker />);
    assert(wrapper.find(".cuke-date-picker").length === 1);
  });
  it.skip("should render find .cuke-date-picker-loading classnames", () => {
    const wrapper = mount(<DatePicker loading />);
    assert(wrapper.find(".cuke-date-picker-loading ").length === 1);
  });
  it("should cannot trigger click event when disabled", () => {
    const onChange = jest.fn();
    const wrapper = shallow(<DatePicker disabled onChange={onChange} />);
    wrapper.find(".cuke-date-picker").simulate("click");
    expect(onChange).not.toHaveBeenCalled();
  });
  it("should render extraFooter", () => {
    const Footer = () => <Button>1</Button>;
    const wrapper = shallow(<DatePicker extraFooter={<Footer />} />);
    assert(wrapper.find(Footer).length === 1);
  });
});
