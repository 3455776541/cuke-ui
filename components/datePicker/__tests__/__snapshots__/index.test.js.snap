// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePicker/> should render a <DatePicker/> components 1`] = `
<div
  class="cuke-date-picker"
>
  <div
    class="cuke-date-picker-inner"
  >
    <input
      class="cuke-input cuke-date-picker-input"
      placeholder=""
      readonly=""
      type="text"
      value="2018-10-30"
    />
    <svg
      class="cuke-date-picker-arrow"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
    >
      <polyline
        points="6 9 12 15 18 9"
      />
    </svg>
  </div>
  <div
    class="cuke-date-picker-content cuke-date-picker-close cuke-ui-no-animate"
  >
    <div
      class="cuke-date-picker-header"
    >
      <span
        class="cuke-date-picker-date"
      >
        2018   10æœˆ
      </span>
      <span
        class="cuke-date-picker-switch"
      >
        <span
          class="cuke-date-picker-switch-group"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
          >
            <polyline
              points="15 18 9 12 15 6"
            />
          </svg>
        </span>
        <span
          class="cuke-date-picker-switch-group"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
          >
            <polyline
              points="9 18 15 12 9 6"
            />
          </svg>
        </span>
      </span>
    </div>
    <div
      class="cuke-date-picker-items"
    >
      <span
        class="cuke-date-picker-item cuke-date-picker-day-title"
      >
        Su
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-day-title"
      >
        Mo
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-day-title"
      >
        Tu
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-day-title"
      >
        We
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-day-title"
      >
        Th
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-day-title"
      >
        Fr
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-day-title"
      >
        Sa
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-last-month"
      >
        30
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        1
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        2
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        3
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        4
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        5
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        6
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        7
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        8
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        9
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        10
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        11
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        12
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        13
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        14
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        15
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        16
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        17
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        18
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        19
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        20
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        21
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        22
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        23
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        24
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        25
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        26
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        27
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        28
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        29
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month cuke-date-picker-selected-date"
      >
        30
      </span>
      <span
        class="cuke-date-picker-item cuke-date-picker-current-month"
      >
        31
      </span>
    </div>
    <div
      class="cuke-date-picker-footer"
    />
  </div>
</div>
`;
